### PR TITLE
Add test to make injectedhelpers compatible with all csharp language versions down to csharp 2

### PR DESCRIPTION
**Pull request by Mobrockers**
Hey @dupdob 

I was working on stryker-mutator/stryker-net#568 and I realized that not all of our injected code might be compatible with all csharp language features so I wrote a test to check the injected code against all languages features. Since csharp lang version 1 does not even support generics I decided not to support that but I rewrote all other syntactic sugar in our injected helpers to be compatible with at least language version 2. It doesn't feel right to me to just push it on to your branch so I'll submit it as a PR instead.

Do you see any issues with the proposed changes?

### CODE REVIEW

File: src/Stryker.Core/Stryker.Core/InjectedHelpers/MutantControl.cs [hunk 1]
```diff
@@ -29,10 +29,10 @@ public static void InitCoverage()
             ActiveMutation = int.Parse(Environment.GetEnvironmentVariable("ActiveMutation") ?? "-1");
             if (channel != null)
             {
-                channel?.Dispose();
+                channel.Dispose();
                 channel = null;
             }
-            var coverageMode = Environment.GetEnvironmentVariable(EnvironmentPipeName)??string.Empty;
+            string coverageMode = Environment.GetEnvironmentVariable(EnvironmentPipeName) ?? string.Empty;
```
Comment (dupdob) [inline src/Stryker.Core/Stryker.Core/InjectedHelpers/MutantControl.cs]: I am a bit surprised: the '??' operator was added as part of C# 6. How come this works?

**Comment by dupdob** [comment]
I am not convinced it is important to support C# 2.0 level, but impact is not important.
My concern is more about framework/BCL compatibility: as of now, the (my) code assume that needed references will be present in the project. If they are not (e.g. System.IO.Pipers), the compilation will fail.

But looking for the proper assemblies is difficult to automate, since we cannot reference every framework version in the project, and I don't know if/how to explore the one that are available programatically ...

File: src/Stryker.Core/Stryker.Core/InjectedHelpers/MutantControl.cs (hunk 1 above)
Comment (Mobrockers) [reply src/Stryker.Core/Stryker.Core/InjectedHelpers/MutantControl.cs]: ? Was added as part of c#6, ?? Has always been there as far as I could find. I was surprised about that as well.

**Comment by Mobrockers** [comment]
We are not planning to support more framework versions, only language versions. I don't necessarily see a huge benefit in supporting csharp2 but I do think there is benefit in limiting the syntax sugar we use in our infected code. CSharp2 support is just a side benefit imo.

**Pull request merged**